redislock.go

```diff
@@ -104,11 +119,12 @@ func (c *Client) Obtain(ctx context.Context, key string, ttl time.Duration, opt
 	}
 }
 
-func (c *Client) obtain(ctx context.Context, key, value string, tokenLen int, ttlVal string) (bool, error) {
-	_, err := luaObtain.Run(ctx, c.client, []string{key}, value, tokenLen, ttlVal).Result()
-	if err == redis.Nil {
-		return false, nil
-	} else if err != nil {
+func (c *Client) obtain(ctx context.Context, keys []string, value string, tokenLen int, ttlVal string) (bool, error) {
+	_, err := luaObtain.Run(ctx, c.client, keys, value, tokenLen, ttlVal).Result()
+	if err != nil {
+		if err.Error() == ErrNotObtained.Error() {
```

Comment:
Sorry, two things:

1. Comparing errors by text is definitely a "no" :), please use `errors.Is`.
2. I don't get it, how can `luaObtain.Run` ever produce an `ErrNotObtained` in the first place?

Reply:
1. Cannot use `errors.Is` since error pointer are not the same.
2. It can't, that's why text compare is required.

I'll update this to simply return false and check for `redis: nil` :)

At some point (during development) it made sense for the script to return distinct errors. But it can only result in a `ErrNotObtained` anyway now.